tests/list_command_test.py

```diff
@@ -1,41 +1,50 @@
 import unittest
 from unittest.mock import patch, MagicMock,Mock
-from pieces.commands.list_command import ListCommand
+from pieces.commands.list_command import ListCommand , PiecesSelectMenu
 from pieces.settings import Settings
-from io import StringIO
+import io
 import sys
 
 class TestListCommand(unittest.TestCase):
-    def test_list_command(self):
-        Settings.startup()
-        for i in ["models","assets","apps"]:
-            # Redirect stdout to a buffer
-            stdout = sys.stdout
-            sys.stdout = StringIO()
-
-            # Call the function that prints to stdout
-            ListCommand.list_command(type = i)
-
-            # Get the output and restore stdout
-            assets = sys.stdout.getvalue()
-            sys.stdout = stdout
-
-            # Check if the function prints a string
-            self.assertIsInstance(assets, str)
-
-            # Check if the string is not empty
-            self.assertTrue(assets)
-
-            assets_list = assets.strip().split('\n')
-            
-            if i == "models":
-                model = assets_list[-1]
-                assets_list = assets_list[:-1] # Remove the line which conatin the current model text
-                self.assertEqual(model,f"Currently using: {Settings.model_name} with uuid {Settings.model_id}")
-            
-            # Check if the string represents a numbered list
-            self.assertTrue(all(line.strip().startswith(str(i+1)) for i, line in enumerate(assets_list) if line.strip()))
+
+    @patch('builtins.print')
+    def test_list_assets(self, mock_print):
+        with patch.object(ListCommand, 'list_assets') as mock_list_assets:
+            ListCommand.list_command(type='assets', max_assets=5)
+            mock_list_assets.assert_called_once_with(5)
+
+    @patch('builtins.print')
+    def test_list_models(self, mock_print):
+        with patch.object(ListCommand, 'list_models') as mock_list_models:
+            ListCommand.list_command(type='models')
+            mock_list_models.assert_called_once()
+
+    def test_list_apps(self):
+        with patch('sys.stdout', new=io.StringIO()) as fake_out:
+            with patch.object(ListCommand, 'list_apps') as mock_list_apps:
+                mock_list_apps.side_effect = lambda: print("1: VS_CODE, 1.17.0, MACOS\n2: PIECES_FOR_DEVELOPERS, 3.0.2, MACOS")
+                ListCommand.list_command(type='apps')
+                
+            output = fake_out.getvalue().strip()
+            self.assertIn('1: VS_CODE, 1.17.0, MACOS', output)
+            self.assertIn('2: PIECES_FOR_DEVELOPERS, 3.0.2, MACOS', output)
```

Comment:
Can we make it a bit more dynamic here (use the platform library)

Reply:
```python
 local_os = platform.system().upper() if platform.system().upper() in ["WINDOWS","LINUX","DARWIN"] else "WEB"
local_os = "MACOS" if local_os == "DARWIN" else local_os
```

Reply:
Addressed in [b4234d7](https://github.com/pieces-app/cli-agent/pull/159/commits/b4234d7e5dd16c355867b8883b1bb705eb4d8b39) commit